File: test_websocket.py
#!/usr/bin/env python3

import asyncio
import websockets
import json


async def test_websocket():
    uri = "ws://localhost:8000/ws"

    try:
        print("🔌 Connecting to WebSocket...")
        async with websockets.connect(uri) as websocket:
            print("✅ Connected!")

            # Send init message
            init_message = {
                "user_id": "test_user_123",
                "thread_id": "test_user_123_abc123def456",
                "init": True,
            }
            print(f"📤 Sending init: {init_message}")
            await websocket.send(json.dumps(init_message))

            # Wait for init response
            response = await websocket.recv()
            print(f"📨 Received: {response}")

            # Send test chat message
            if "ok" in response:
                chat_message = {"message": "Hello, this is a test!"}
                print(f"📤 Sending chat: {chat_message}")
                await websocket.send(json.dumps(chat_message))

                # Receive responses
                for i in range(10):  # Listen for up to 10 responses
                    try:
                        response = await asyncio.wait_for(websocket.recv(), timeout=2.0)
                        print(f"📨 Response {i+1}: {response}")

                        # Stop if we get final message
                        if "assistant_message" in response:
                            break

                    except asyncio.TimeoutError:
                        print("⏰ No more messages received")
                        break

    except Exception as e:
        print(f"❌ Error: {e}")


if __name__ == "__main__":
    asyncio.run(test_websocket())

File: test_websocket.js
#!/usr/bin/env node

const WebSocket = require('ws');

const ws = new WebSocket('ws://localhost:8000/ws');

ws.on('open', function open() {
    console.log('✅ Connected to WebSocket');

    // Send init message
    ws.send(JSON.stringify({
        user_id: 'test_user',
        thread_id: 'test_thread',
        init: true
    }));
});

ws.on('message', function message(data) {
    console.log('📨 Received:', data.toString());

    // After successful init, send a test message
    if (data.toString().includes('initialized')) {
        console.log('✅ Sending test message...');
        ws.send(JSON.stringify({
            message: 'Hello, this is a test message!'
        }));
    }
});

ws.on('close', function close() {
    console.log('❌ Connection closed');
});

ws.on('error', function error(err) {
    console.error('🚨 WebSocket error:', err);
});

// Close connection after 30 seconds
setTimeout(() => {
    console.log('⏰ Closing connection...');
    ws.close();
}, 30000);